fastembed/text/onnx_text_model.py

```diff
@@ -104,23 +109,60 @@ def _embed_documents(
         if parallel == 0:
             parallel = os.cpu_count()
 
-        if parallel is None or is_small:
+        if is_small:
+            model = cls(model_name=model_name, cache_dir=cache_dir, providers=providers, **kwargs)
             for batch in iter_batch(documents, batch_size):
-                yield from self._post_process_onnx_output(self.onnx_embed(batch))
+                yield from model._post_process_onnx_output(model.onnx_embed(batch))
         else:
-            start_method = (
-                "forkserver" if "forkserver" in get_all_start_methods() else "spawn"
-            )
-            params = {"model_name": model_name, "cache_dir": cache_dir, **kwargs}
-            pool = ParallelWorkerPool(
-                parallel, self._get_worker_class(), start_method=start_method
-            )
+            use_multi_gpu = providers and "CUDAExecutionProvider" in providers and device_ids
```

Comment:
I thought we discussed that usage of `cuda=True` param should prevent exactly this piece if code. Was this reconsidered?